### ISSUE TITLE
old title (wrong one): transaction list -h; new title: transaction list DECK

### ISSUE THREAD

**Issue by buhtignew** [slimcoin-project/pacli#54]
Maybe it could make sense ~~editing the following two lines in `transaction list -h`:
`It can be a label or a deck ID.` into `It can be a label (global name if no local label is set) or a deck ID.`
`DECK can be a label or a deck ID.` into `DECK can be a label (global name if no local label is set) or a deck ID.`
Or in alternative we can~~ let the code search both in local labels and global names ~~, which would be even more comfortable~~.

**Comment by d5000** [comment]
I'll think about it, this is an easy change. My problem with that approach is: I generally don't want people to confuse global/local names so I tend to try to separate the commands for them.

For example one challenge is: Which name should have priority, if a local name has a different deck ID as value than a global name? If we can come up with good rules for this ambiguity, I can change it. (I would opt for the local name, because that's based on a conscient decision of the user).

**Comment by buhtignew** [comment]
I think the local label is the priority because they can help people getting rid of the same global names.

Thus if someone inputs a label the first thing the code should check whether there is such a local label, if there is none the global name is checked.
If there are same global names the message should be something like:
"Error: multiple decks with the same global name GLOBAL_NAME, please consider setting the local label for the deck you'd like to use in this case (pacli deck set LABEL DECKID command) or use the desired deck_id"

Can it work for you?

**Comment by d5000** [comment]
Search for global deck names has been added now in commit c6d4bc9 (pushed also the other commit with minor fixes). Affects all commands where deck names/labels can be entered as an argument.

The priority is as follows (basically like your suggestion):
1) local label
2) global name
3) deck ID

In the case of more than one deck with the same name, the priority is:
1) issue time (this is basically the block height) -> first deck published is used
2) TXID -> lower TXID is used.

I could have outputted an error message in this case. But if the process is completely deterministic then there will be a small advantage for the first deck published with a certain name. This could incentive the use of decks, if we look at the Ordinals ecosystem, e.g. for memecoins.

You can test the priority mechanism with the decks called `conflicting` which are in the same block.

Can be closed if everything works.

**Comment by buhtignew** [comment]
> In the case of more than one deck with the same name, the priority is:
    issue time (this is basically the block height) -> first deck published is used
    TXID -> lower TXID (lower hexadecimal number, so those with 0 at the start are first) is used.

Great idea!
I think this can be closed now.

**Comment by buhtignew** [comment]
Maybe we should consider to put the DeckID on the first place of our priority list because someone can create a token with the global name of another token's DeckID and people who'd need to get rid of those ambiguities may have some issue with that.
What do you think?

**Comment by d5000** [comment]
Yes, that makes sense, so in case of any ambiguity you're always safe to use the Deck ID.

I just saw the function which transforms labels into Deck IDs is already implemented this way: if the given string is a transaction ID, then it will return it first (this also works for proposals in dPoD tokens).

I added an additional check: if the deck id is not found in the existing decks, then it will continue with the other branches (searching for a label or deck). Commit 7446f9b
